mmengine/dataset/dataset_wrapper.py

```diff
@@ -45,13 +48,37 @@ def __init__(self,
                 raise TypeError(
                     'elements in datasets sequence should be config or '
                     f'`BaseDataset` instance, but got {type(dataset)}')
+        if ignore_keys is None:
+            self.ignore_keys = []
+        elif isinstance(ignore_keys, str):
+            self.ignore_keys = [ignore_keys]
+        elif isinstance(ignore_keys, list):
+            self.ignore_keys = ignore_keys
+        else:
+            raise TypeError('ignore_keys should be a list or str, '
+                            f'but got {type(ignore_keys)}')
+
+        metainfos = [dataset.metainfo.keys() for dataset in self.datasets]
+        meta_keys = set(metainfos[0])
```

Comment:
```suggestion
        metainfos = [set(dataset.metainfo.keys()) for dataset in self.datasets]
        meta_keys = metainfos[0]
```

`dict.keys()` will not return a repeated result.